pkg/appgw/configbuilder.go

```diff
@@ -254,7 +254,7 @@ func generateListenerID(ingress *networking.Ingress, rule *networking.IngressRul
 	}
 
 	extendedHostNames, err := annotations.GetHostNameExtensions(ingress)
-	if err != nil {
+	if err != nil && controllererrors.IsErrorCode(err, controllererrors.ErrorInvalidContent) {
```

Comment:
`annotations.GetHostNameExtensions()` seems to only return errors with code `controllererrors.ErrorMissingAnnotation`. What does this error code check do?

Reply:
changed to `if err != nil && !controllererrors.IsErrorCode(err, controllererrors.ErrorMissingAnnotation) {`